### PR TITLE
Bugfix for spring layouts with empty list of fixed nodes

### PR DESCRIPTION
**Pull request by iosonofabio**
Thanks for maintaining this awesome library!

Spring layout accepts a list of "fixed" nodes that won't move from (required) initial positions.

If that list is empty (an edge but acceptable case), networkx currently freaks out. This PR fixes the bug by telling numpy, which is used to create the array of fixed node indices, that this array is of type int no matter whether it's empty or not.

I found this while testing [ilayoutx](https://github.com/fabilab/ilayoutx) since I'm covering these cases there. Similar edge cases might be found in other layout functions, but that's untested for now.

Do you want me to add some kind of unit test for this?

### CODE REVIEW

**Comment by amcandio** [comment]
Thanks, can you show the error you are seeing? +1 on the unit test!

**Comment by iosonofabio** [comment]
No problem and thanks for the quick reply.

The error is numpy trying to use the fixed array to index delta_pos and thinking the dtype is neither int nor bool.

It happens when arrays are initialised without a dtype (like in the current code) and numpy cannot guess the dtype (because the array is empty), so it defauls to object.

Which file would the unit test go into?

**Comment by amcandio** [comment]
https://github.com/networkx/networkx/blob/main/networkx/drawing/tests/test_layout.py here seems like the best candidate

**Comment by iosonofabio** [comment]
Cheers, will add in a couple hours

**Comment by iosonofabio** [comment]
I added a unit test for this one case, should be ready for merging now

File: networkx/drawing/tests/test_layout.py [hunk 1]
```diff
@@ -206,6 +206,10 @@ def test_fixed_node_fruchterman_reingold(self):
         npos = nx.spring_layout(self.bigG, pos=pos, fixed=[(0, 0)])
         for axis in range(2):
             assert pos[(0, 0)][axis] == pytest.approx(npos[(0, 0)][axis], abs=1e-7)
+        # Empty fixed list
```
Comment (rossbar) [inline networkx/drawing/tests/test_layout.py]:
```suggestion
        # Empty fixed list - see gh-8446
```

**Review by rossbar — APPROVED** [review]
Nice corner case! Thanks @iosonofabio and @amcandio !
